### PR TITLE
LPS-159341 Frontend tests

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by veroglez** [comment]
ci:test:sf

**Comment by veroglez** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/common/components/LengthField.test.js [hunk 1]
```diff
@@ -83,12 +96,53 @@ describe('LengthField', () => {
...
 		renderLengthField();
-		const input = screen.getByLabelText('Length Field');
+		const input = screen.getByLabelText('length-field');
 
 		userEvent.type(input, 'auto');
 
 		expect(input).toHaveValue(null);
 	});
+
+	describe('If LengthField is part of a Select', () => {
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/test/page_editor/common/components/LengthField.test.js:116]: We should use a valid sentence here maybe describe LengthField when it is part of a Select field?

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/common/components/LengthField.test.js [hunk 2]
```diff
@@ -83,12 +96,53 @@ describe('LengthField', () => {
...
+
+		it('focuses the input when the currently option is custom and a other unit is selected', () => {
+			renderLengthField({field, value: 'calc(12px - 3px)'});
+
+			userEvent.click(screen.getByText('%'));
+
+			expect(screen.getByLabelText('length-field')).toHaveFocus();
+		});
+
+		it('does not save the value and keeps the previous value when the input is cleared', async () => {
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/test/page_editor/common/components/LengthField.test.js:132]: `async` is not needed here. Also can't we not clear the value? :o

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/components/fragment-configuration-fields/AdvancedSelectField.test.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+				typeOptions: {
+					showLengthField: true,
+				},
+			},
+			value: 'mystyle',
+		});
+
+		expect(screen.getByTitle('select-units')).toBeInTheDocument();
+	});
+});
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/components/fragment-configuration-fields/AdvancedSelectField.test.js:203]: There is a bunch querySelector in this test, can't we use any of the screen function to grab the reference?

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/components/fragment-configuration-fields/AdvancedSelectField.test.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+	},
+};
+
+const renderAdvancedSelectField = ({
+	field = FIELD,
+	onValueSelect = () => {},
+	value = '',
+} = {}) => {
+	const state = {
+		permissions: {UPDATE: true},
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/components/fragment-configuration-fields/AdvancedSelectField.test.js:86]: I think we can also add a test here, that check that the detach token button is not present if the user doesn't have the update permission

**Review by victorg1991 — COMMENTED** [review]
Super clean tests 👏  Just added a couple of comments

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/common/components/LengthField.test.js (hunk 2 above)
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/common/components/LengthField.test.js:132]: yes, that's it. When the value is deleted the previous one is kept.

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/components/fragment-configuration-fields/AdvancedSelectField.test.js (hunk 3 above)
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/components/fragment-configuration-fields/AdvancedSelectField.test.js:205]: I've changed a couple of tests to remove the querySelector. I have added an `aria-label` to the select, so now I use the `getByLabelText` to select the input/select.

**Comment by victorg1991** [comment]
Nice! 👏

**Comment by victorg1991** [comment]
ci:forward

**Comment by veroglez** [comment]
ci:forward

**Comment by ozysouza** [comment]
ci:forward

**Comment by ozysouza** [comment]
ci:forward

**Comment by ozysouza** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/121540

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**